src/Query/Filter/DateField.ts

```diff
@@ -99,15 +99,27 @@ export abstract class DateField extends Field {
      */
     protected buildFilterFunction(fieldKeyword: string, fieldDates: DateRange): FilterFunction {
         let dateFilter: DateFilterFunction;
-        if (fieldKeyword === 'before') {
-            dateFilter = (date) => (date ? date.isBefore(fieldDates.start) : this.filterResultIfFieldMissing());
-        } else if (fieldKeyword === 'after') {
-            dateFilter = (date) => (date ? date.isAfter(fieldDates.end) : this.filterResultIfFieldMissing());
-        } else {
-            dateFilter = (date) =>
-                date
-                    ? date.isSameOrAfter(fieldDates.start) && date.isSameOrBefore(fieldDates.end)
-                    : this.filterResultIfFieldMissing();
+        switch (fieldKeyword) {
```

Comment:
I think this has become complicated enough that an explanatory comment in each case would be really helpful.

Something very vaguely like:

```ts
// 'on or before'/'in or before' a date range uses the end of the range
// as the search limit, so that it matches every date in the 
// inclusive date range, and all dates before the range.
```

Reply:
Agree. Added comments as yours, just splitted the difference between <date> and <date range>.